app/src/features/surveys/observations/observations-table/configure-columns/components/measurements/useConfigureMeasurementColumns.tsx

```diff
@@ -0,0 +1,103 @@
+import { getSurveySessionStorageKey, SIMS_OBSERVATIONS_MEASUREMENT_COLUMNS } from 'constants/session-storage';
+import { useObservationsTableContext, useSurveyContext } from 'hooks/useContext';
+import { CBMeasurementType } from 'interfaces/useCritterApi.interface';
+import { useCallback } from 'react';
+
+/**
+ * Functions for measurement column configuration.
+ *
+ * @return {*}
+ */
+export const useConfigureMeasurementColumns = () => {
+  const surveyContext = useSurveyContext();
+  const observationsTableContext = useObservationsTableContext();
+
+  /**
+   * Handles the addition of measurement columns to the table.
+   *
+   * @param {CBMeasurementType[]} measurements
+   * @return {*}
+   */
+  const onAddMeasurementColumns = useCallback(
+    (measurementColumnsToAdd: CBMeasurementType[]) => {
+      if (!measurementColumnsToAdd?.length) {
+        return;
+      }
+
+      // Add the measurement columns to the table context
+      observationsTableContext.setMeasurementColumns((currentColumns) => {
+        const newColumns = measurementColumnsToAdd.filter(
+          (columnToAdd) =>
+            !currentColumns.find(
+              (currentColumn) => currentColumn.taxon_measurement_id === columnToAdd.taxon_measurement_id
+            )
+        );
+
+        // Store all measurement definitions in local storage
+        sessionStorage.setItem(
+          getSurveySessionStorageKey(surveyContext.surveyId, SIMS_OBSERVATIONS_MEASUREMENT_COLUMNS),
+          JSON.stringify([...currentColumns, ...newColumns])
+        );
+
+        return [...currentColumns, ...newColumns];
+      });
+    },
+    [observationsTableContext, surveyContext.surveyId]
+  );
+
+  /**
+   * Handles the removal of measurement columns from the table.
+   *
+   * @param {string[]} measurementColumnsToRemove The `field` names of the columns to remove. For measurementColumnsToAdd, this is
+   * the `taxon_measurement_id`.
+   */
+  const onRemoveMeasurementColumns = useCallback(
+    (measurementColumnsToRemove: string[]) => {
+      // Delete the measurement columns from the database
+      observationsTableContext.deleteObservationMeasurementColumns(measurementColumnsToRemove, () => {
+        // Remove the measurement columns from the table context
+        observationsTableContext.setMeasurementColumns((currentColumns) => {
+          const remainingColumns = currentColumns.filter(
+            (currentColumn) => !measurementColumnsToRemove.includes(currentColumn.taxon_measurement_id)
+          );
+
+          // Store all remaining measurement definitions in local storage
+          sessionStorage.setItem(
```

Comment:
Once 572 goes in, you can use `usePersistentState` hook if you want

Reply:
Ill look at incorporating it in a separate PR, just to get his chunk in sooner.

I did start to drop it in, but in a few places I do some filtering of the state (which could maybe be removed, but not sure), that will require a bit more testing to make sure I don't accidentally break anything.